api/v1/installation_types.go

```diff
@@ -129,17 +129,7 @@ type InstallationSpec struct {
 type TyphaAffinity struct {
 	// NodeAffinity describes node affinity scheduling rules for typha.
 	// +optional
-	NodeAffinity *PreferredNodeAffinity `json:"nodeAffinity,omitempty"`
-}
-
-// PreferredNodeAffinity is similar to an affinity except it only exposes the Preferred scheduling option,
-// which ensures a pod will still be scheduled even when all the nodes it matches are unschedulable.
-type PreferredNodeAffinity struct {
-	// The scheduler will prefer to schedule pods to nodes that satisfy
-	// the affinity expressions specified by this field, but it may choose
-	// a node that violates one or more of the expressions.
-	// +optional
-	PreferredDuringSchedulingIgnoredDuringExecution []v1.PreferredSchedulingTerm `json:"preferredDuringSchedulingIgnoredDuringExecution,omitempty"`
```

Comment:
@caseydavenport do you agree that instead of explicitly putting the RequiredDuring for ProviderAKS, now we should default the field in the Installation resource?

Reply:
> do you agree that instead of explicitly putting the RequiredDuring for ProviderAKS, now we should default the field in the Installation resource?

Yeah, I think that is the correct way to do this.

Reply:
I don't think it's a good idea to move this to default, since if we do that, then user is able to override this, which in turn means that if someone will override it in Azure, it will re-enable scheduling on those nodes which cannot be scheduled. Current behavior will be better - since this override will take precedence.

Reply:
@tmjd I moved it to default. I think it's ready for re-review.